protocol.md

```diff
@@ -870,6 +902,12 @@ export interface TextDocumentClientCapabilities {
 ```typescript
 interface ClientCapabilities {
 	/**
+	 * The client supports receiving the result solely through $/partialResult notifications for requests from the client to the server.
+	 * If true, the server should send all data through $/partialResult and send null as the result in the final response.
```

Comment:
@felixfbecker this is were out of order can happen https://sourcegraph.com/github.com/sourcegraph/jsonrpc2@7ea7528660e2e6a2b0ee6f4ff23d0d99bc4c2a55/-/blob/jsonrpc2.go#L480:32
We receive the messages over the wire in the correct order, but each message gets its own goroutine. What order those run in are dependent on the scheduler/os. I assume this sort of way of handling requests is common across languages. In this case we would have to modify our jsonrpc2 library to avoid this problem. It is actually one of the reasons why I did a 1s sleep between notifications, to hopefully avoid the race :)

It honestly is starting to feel like the correct place for this to exist is in the jsonrpc2 spec, but that seems less practical. Or we could start going down the road @nicksnyder suggested and do it with our own IDs.

Whatever we do, I'm keen to just keep moving with our current implementation we have and getting that into production. Whatever we decide won't actually affect the user visible feature and is possible to change.